src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -150,18 +181,32 @@ loadMesh( Path const & filePath )
   return loadedMesh;
 }
 
+vtkNew< vtkCellArray > GetCellArray( vtkDataSet & mesh ) // replaces GetCells() that exist only in vtkUnstructuredGrid
+{
+  vtkNew< vtkCellArray > cells;
+  if( mesh.IsA( "vtkStructuredGrid" ) || mesh.IsA( "vtkUnstructuredGrid" ))
+  {
+    int numCell = mesh.GetNumberOfCells();
+    for( int c = 0; c < numCell; c++ )
+    {
+      cells->InsertNextCell( mesh.GetCell( c ));
+    }
+  }
+  return cells;
+}
```

Comment:
It would be nice to avoid constructing a new `vtkCellArray` when the grid is unstructured and already contains one. I suggest the following:
  * change return type to `vtkSmartPointer< vtkCellArray >`
  * have a branch that is executed when `mesh.IsA( "vtkUnstructuredGrid" )` is true and simply returns `vtkUnstructuredGrid::SafeDownCast( &mesh )->GetCells()`
  * do what you're already doing for all other types of input datasets.

Reply:
I changed the function with your suggestions in commit 7b16966